tools/w4mconcatenate/concat.xml

```diff
@@ -0,0 +1,275 @@
+   <param name="metadata_1" type="data" label="Metadata file 1" format="tabular"/>
+   <param name="metadata_2" type="data" label="Metadata file 2" format="tabular"/>
+  
+   <param name="type" type="select" display="radio" label="Type of metadata">
+    <option value="sample">Sample metadata</option>
+    <option value="variable">Variable metadata</option>
+   </param>
+  
+   <param name="concatenation" type="select" display="radio" label="Type of concatenation">
+    <option value="unique">Unique</option>
+    <option value="intersection">Intersection</option>
+    <option value="union">Union</option>
+   </param>
+  
+   <conditional name="choice">
+    <param name="choice_keep" type="select" display="radio" label="Keep all or just one">
+     <option value="oui">Keep all</option>
+     <option value="non">Keep one</option>
+    </param>
+    <when value="non">
+     <param name="keep" type="select" display="radio" label="Which Metadata to keep">
+      <option value="1">Metadata 1</option>
+      <option value="2">Metadata 2</option>
+     </param>
+    </when>
+    <when value="oui">
+    </when>
+   </conditional>
+  
+  <param name="tab1" type="text" label="Suffix for Metadata 1" value="Tab1"/>
+  <param name="tab2" type="text" label="Suffix for Metadata 2" value="Tab2"/>
+  
+  </inputs>
+  
+  
+  
+  <outputs>
+  
+   <data name="dataMatrix_1_out" label="Concat_${dataMatrix_1.name}" format="tabular"></data>
+   <data name="dataMatrix_2_out" label="Concat_${dataMatrix_2.name}" format="tabular"></data>
+   <data name="metadata_out" label="Concat_${metadata_1.name}" format="tabular"></data>
+  
+  </outputs>
+  
+  <tests>
+   <test>
+    <param name="dataMatrix_1" value="Input_Unique_Test_1-2-3-4-5_DM1.txt"/>
+    <param name="dataMatrix_2" value="Input_Unique_Test_1-2-3_DM2.txt"/>
+    <param name="metadata_1" value="Input_Unique_Test_1_SM1.txt"/>
+    <param name="metadata_2" value="Input_Unique_Test_1_SM2.txt"/>
+    <param name="concatenation" value="unique"/>
+    <param name="choice_keep" value="oui"/>
+    <param name="tab1" value="Tab1"/>
+    <param name="tab2" value="Tab2"/>
+  
+    <output name="dataMatrix_1_out" file="Output_Attendu_Unique_Test_1-2-3_DM1.txt"/>
+    <output name="dataMatrix_2_out" file="Output_Attendu_Unique_Test_1-2-3_DM2.txt"/>
+    <output name="metadata_out" file="Output_Attendu_Unique_Test_1_Metadata.txt"/>
+  
+  
+   </test>
+  
+  
+   <test expect_failure="true">
+    <param name="dataMatrix_1" value="Input_Unique_Test_1-2-3-4-5_DM1.txt"/>
+    <param name="dataMatrix_2" value="Input_Unique_Test_4_DM2.txt"/>
+    <param name="metadata_1" value="Input_Unique_Test_4_SM1.txt"/>
+    <param name="metadata_2" value="Input_Unique_Test_4_SM2.txt"/>
+    <param name="concatenation" value="unique"/>
+    <param name="choice_keep" value="oui"/>
+    <param name="tab1" value="Tab1"/>
+    <param name="tab2" value="Tab2"/>
+  
+  
+  
+  
+   </test>
+  
+  </tests>
+  
+  
+  
+  
```

Comment:
Can you reduce the number of empty lines?

Reply:
Corrected in 403025a